homeassistant/components/fujitsu_hvac/climate.py

```diff
@@ -0,0 +1,205 @@
+"""Support for Fujitsu HVAC devices that use the Ayla Iot platform."""
+from asyncio import gather
+from contextlib import suppress
+from typing import Any
+
+from ayla_iot_unofficial import AylaAuthError
+from ayla_iot_unofficial.fujitsu_hvac import Capability, FujitsuHVAC
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_HALVES, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    API,
+    DOMAIN,
+    FUJI_TO_HA_FAN,
+    FUJI_TO_HA_HVAC,
+    FUJI_TO_HA_SWING,
+    HA_TO_FUJI_FAN,
+    HA_TO_FUJI_HVAC,
+    HA_TO_FUJI_SWING,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up one Fujitsu HVAC device."""
+    api = hass.data[DOMAIN][API]
+
+    devices = await api.async_get_devices()
+    devices = list(filter(lambda x: isinstance(x, FujitsuHVAC), devices))
+    await gather(*[dev.async_update() for dev in devices])
+
+    async_add_entities([FujitsuHVACDevice(dev) for dev in devices])
```

Comment:
```suggestion
    async_add_entities(
    [FujitsuHVACDevice(dev)
    if isinstance(dev, FujitsuHVAC)
    for dev in await api.async_get_devices()], True)
```
Formatting isn't right, but this code does exactly the same as the code above :)

Reply:
Except it calls async_update() after __init__ of each dev but some of the elements of device_info are fetched through async_update().